File: eCar/eCar.Model/Model/Route.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity.Spatial;
using System.Text;
using eCar.Model.DTO;
using Microsoft.Spatial;
using Microsoft.SqlServer.Types;
using NetTopologySuite.Geometries;

namespace eCar.Model.Model
{
    public class Route
    {
        public int Id { get; set; }

        public PointDTO? SourcePoint { get; set; }

        public PointDTO? DestinationPoint { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int? Duration { get; set; }
        public decimal? NumberOfKilometars { get; set; }
        public decimal? FullPrice { get; set; }
        public bool? Paid { get; set; }
        public string? Status { get; set; }
        public int DriverID { get; set; }
        public virtual Driver Driver { get; set; } = null!;
        public int ClientId { get; set; }
        public virtual Client Client { get; set; } = null!;

       
    }
}

File: eCar/UI/ecar_mobile/lib/models/Route/route.dart
import 'package:ecar_mobile/models/Driver/driver.dart';
import 'package:ecar_mobile/utils/PointDTO/point_dto.dart';
import 'package:ecar_mobile/models/Client/client.dart' as Model;

import 'package:json_annotation/json_annotation.dart';

part 'route.g.dart';

@JsonSerializable()
class Route {
  int? id;
  Pointdto? sourcePoint;
  Pointdto? destinationPoint;
  String? status;
  DateTime? startDate;
  DateTime? endDate;
  int? duration;
  double? numberOfKilometars;
  double? fullPrice;
  bool? paid;
  int? driverID;
  int? clientId;
  Driver? driver;
  Model.Client? client;
  Route(
      {this.id,
      this.sourcePoint,
      this.destinationPoint,
      this.status,
      this.startDate,
      this.endDate,
      this.duration,
      this.numberOfKilometars,
      this.fullPrice,
      this.paid,
      this.driverID,
      this.clientId,
      this.driver,
      this.client});

  factory Route.fromJson(Map<String, dynamic> json) => _$RouteFromJson(json);

  Map<String, dynamic> toJson() => _$RouteToJson(this);
}
